types/yargs-parser/index.d.ts

```diff
@@ -0,0 +1,56 @@
+// Type definitions for yargs-parser 10.1
+// Project: https://github.com/yargs/yargs-parser#readme
+// Definitions by: Miles Johnson <https://github.com/milesj>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+import { Arguments } from 'yargs';
+
+export { Arguments };
+
+export interface Configuration {
+    'boolean-negation': boolean;
+    'camel-case-expansion': boolean;
+    'combine-arrays': boolean;
+    'dot-notation': boolean;
+    'duplicate-arguments-array': boolean;
+    'flatten-duplicate-arrays': boolean;
+    'negation-prefix': string;
+    'parse-numbers': boolean;
+    'populate--': boolean;
+    'set-placeholder-key': boolean;
+    'short-option-groups': boolean;
+}
+
+export interface Options {
+    alias?: { [key: string]: string | string[] };
+    array?: string[];
+    boolean?: string[];
+    config?: string | string[] | { [key: string]: boolean };
+    configuration?: Partial<Configuration>;
+    coerce?: { [key: string]: (arg: any) => any };
+    count?: string[];
+    default?: { [key: string]: any };
+    envPrefix?: string;
+    narg?: { [key: string]: number };
+    normalize?: string[];
+    string?: string[];
+    number?: string[];
+    '--'?: boolean;
+}
+
+export interface DetailedArguments {
+    argv: Arguments;
+    error: Error | null;
+    aliases: { [alias: string]: string[] };
+    newAliases: { [alias: string]: boolean };
+    configuration: Configuration;
+}
+
+interface YargsParser {
+    (argv: string | string[], opts?: Options): Arguments;
+    detailed(argv: string | string[], opts?: Options): DetailedArguments;
+}
+
+declare const parse: YargsParser;
+export default parse;
```

Comment:
This should be `export = parse` and not a default export, as the library does a [namespace assignment](https://github.com/yargs/yargs-parser/blob/master/index.js#L833) and does not actually export a member named "default". To import it as a default, it should be used with the `esModuleInterop` and `allowSyntheticDefaultImports` compiler options.

Reply:
@weswigham Done. Had to switch to namespaces for this to work, otherwise it kept throwing errors. However, I'm still getting an error on the `Arguments` type import. Is this possible?

I feel like it is since I can do this: `import { Arguments } from 'yargs';` for the other package.

Reply:
That's confusing, as `yargs` has `Arguments` IN the namespace, yet you can named import it. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/yargs/index.d.ts#L230

Reply:
@weswigham The error is just `ERROR: 1:17  expect  TypeScript@next compile error: 
Module '"/Users/miles_johnson/Sites/DefinitelyTyped/types/yargs-parser/index"' has no exported member 'Arguments'.`

Reply:
@weswigham What about now?